test/common/upstream/sds_test.cc

```diff
@@ -304,4 +304,27 @@ TEST_F(SdsTest, HealthChecker) {
   EXPECT_EQ(4UL, cluster_->healthyHostsPerZone()[2].size());
 }
 
+TEST_F(SdsTest, Failure) {
+  setupRequest();
+  cluster_->initialize();
+
+  std::string bad_response_json = R"EOF(
+  {
+    "hosts" : {}
+  }
+  )EOF";
+
+  Http::MessagePtr message(new Http::ResponseMessageImpl(
+      Http::HeaderMapPtr{new Http::TestHeaderMapImpl{{":status", "200"}}}));
+  message->body(Buffer::InstancePtr{new Buffer::OwnedImpl(bad_response_json)});
+
+  EXPECT_CALL(*timer_, enableTimer(_));
+  callbacks_->onSuccess(std::move(message));
+
+  EXPECT_CALL(*timer_, enableTimer(_));
```

Comment:
You should be calling timer_->callback_() before this, but I would just delete this second case. It's not relevant to what you are testing.